wait/wg.go

```diff
@@ -2,14 +2,31 @@ package wait
 
 import "sync"
 
+type PanicHandler interface {
+	HandlePanic()
+}
+
 type Group struct {
-	wg sync.WaitGroup
+	wg           sync.WaitGroup
+	panicHandler PanicHandler
+}
+
+func (wg *Group) SetPanicHandler(panicHandler PanicHandler) {
```

Comment:
Can we have this as part of the constructor so it's not possible to change this after it is created?